### DISCUSSION TITLE
Compute rotor, stator masses

### DISCUSSION THREAD

**Discussion by Lavanya-Bala** [Eomys/pyleecan#439] [Q&A]
Hello, I am trying to print the stator and rotor masses using the Comp_masses().
It throws the following error, AttributeError: 'MachineUD' object has no attribute 'rotor'. But I have defined the stator and rotor in the machineUD. Could you please let me know how to go about this or if there is a test implementation of this that I could check to see where I am going wrong ?

**Comment by SebGue** [comment]
Hello, could you be more specific, i.e. is there a file and a line number mentioned in the error message?

**Reply by Lavanya-Bala** [reply]
Apologies, this is the line
File "\pyleecan\Methods\Machine\Machine\comp_masses.py", line 33, in comp_masses
Mrot = self.rotor.comp_masses()

**Reply by BonneelP** [reply]
Hello,

The comp_masses is a rather old method and we never have updated it after introducing the MachineUD. With the recent modifications it will be easy to update (a loop on the lamination list and a dict with the new lamination label), it will include it in the current PR #437 and I will release a new version by the end of the week. Thank you for finding this issue :)

Best regards,
Pierre

**Reply by BonneelP** [reply]
Done in 5e27d8a. I took the opportunity to refresh the labels of the dictionary to have something more consistent and clear (Msta => Stator-0, Msha => Shaft, Mtot => All...). What do you think of these new labels ? (I can revert some label for retro-compatibility but then what should we use instead of Msta, Msta0 ?)
